spotinst/ocean_aws_launch_configuration/fields_spotinst_ocean_aws_launch_configuration.go

```diff
@@ -767,3 +910,147 @@ func flattenInstanceMetadataOptions(instanceMetadataOptions *aws.InstanceMetadat
 
 	return []interface{}{result}
 }
+func flattenBlockDeviceMappings(bdms []*aws.ClusterBlockDeviceMappings) []interface{} {
+	result := make([]interface{}, 0, len(bdms))
+
+	for _, bdm := range bdms {
+		m := make(map[string]interface{})
+		m[string(DeviceName)] = spotinst.StringValue(bdm.DeviceName)
+		if bdm.EBS != nil {
+			m[string(Ebs)] = flattenEbs(bdm.EBS)
+		}
+
+		result = append(result, m)
+	}
+	return result
+
+}
+func flattenEbs(ebs *aws.ClusterEBS) []interface{} {
+
+	elasticBS := make(map[string]interface{})
+	elasticBS[string(DeleteOnTermination)] = spotinst.BoolValue(ebs.DeleteOnTermination)
+	elasticBS[string(Encrypted)] = spotinst.BoolValue(ebs.Encrypted)
+	elasticBS[string(IOPS)] = spotinst.IntValue(ebs.IOPS)
+	elasticBS[string(KMSKeyID)] = spotinst.StringValue(ebs.KMSKeyID)
+	elasticBS[string(SnapshotID)] = spotinst.StringValue(ebs.SnapshotID)
+	elasticBS[string(VolumeType)] = spotinst.StringValue(ebs.VolumeType)
+	elasticBS[string(VolumeSize)] = spotinst.IntValue(ebs.VolumeSize)
+	elasticBS[string(Throughput)] = spotinst.IntValue(ebs.Throughput)
+	if ebs.DynamicVolumeSize != nil {
+		elasticBS[string(DynamicVolumeSize)] = flattenDynamicVolumeSize(ebs.DynamicVolumeSize)
+	}
+
+	return []interface{}{elasticBS}
+}
+func flattenDynamicVolumeSize(dvs *aws.ClusterDynamicVolumeSize) interface{} {
+
+	DynamicVS := make(map[string]interface{})
+	DynamicVS[string(BaseSize)] = spotinst.IntValue(dvs.BaseSize)
+	DynamicVS[string(Resource)] = spotinst.StringValue(dvs.Resource)
+	DynamicVS[string(SizePerResourceUnit)] = spotinst.IntValue(dvs.SizePerResourceUnit)
+
+	return []interface{}{DynamicVS}
+}
+func expandBlockDeviceMappings(data interface{}) ([]*aws.ClusterBlockDeviceMappings, error) {
+
+	list := data.([]interface{})
+	bdms := make([]*aws.ClusterBlockDeviceMappings, 0, len(list))
+
+	for _, v := range list {
+		attr, ok := v.(map[string]interface{})
+		bdm := &aws.ClusterBlockDeviceMappings{}
+
+		if !ok {
+			continue
+		}
+
+		if v, ok := attr[string(DeviceName)].(string); ok && v != "" {
+			bdm.SetDeviceName(spotinst.String(v))
+		}
+
+		if r, ok := attr[string(Ebs)]; ok {
+			if ebs, err := expandEbs(r); err != nil {
+				return nil, err
+			} else {
+				bdm.SetEBS(ebs)
+			}
+		}
+		bdms = append(bdms, bdm)
+	}
+	return bdms, nil
+}
+func expandEbs(data interface{}) (*aws.ClusterEBS, error) {
+
+	ebs := &aws.ClusterEBS{}
+	list := data.([]interface{})
+
+	if list == nil || list[0] == nil {
```

Comment:
re-check the logic here

Reply:
Checked, to make successful API call EBS should has inner paramters.